rs/sns/integration_tests/src/timers.rs

```diff
@@ -211,6 +222,97 @@ fn test_swap_reset_timers() {
     }
 }
 
+#[test]
+fn test_governance_reset_timers() {
+    let state_machine = state_machine_builder_for_sns_tests().build();
+
+    // Install the Governance canister.
+    let wasm = build_governance_sns_wasm().wasm;
+
+    let args = Encode!(&governance_proto()).unwrap();
+    let canister_id = state_machine
+        .install_canister(wasm.clone(), args, None)
+        .unwrap();
+
+    // Helpers.
+    let get_timers = || {
+        let payload = Encode!(&governance_pb::GetTimersRequest {}).unwrap();
+        let response = state_machine
+            .execute_ingress(canister_id, "get_timers", payload)
+            .expect("Unable to call get_timers on the Governance canister");
+        let response = Decode!(&response.bytes(), governance_pb::GetTimersResponse).unwrap();
+        response.timers
+    };
+
+    let last_spawned_timestamp_seconds = {
+        let last_reset_timestamp_seconds = assert_matches!(get_timers(), Some(governance_pb::Timers {
+            last_reset_timestamp_seconds: Some(last_reset_timestamp_seconds),
+            last_spawned_timestamp_seconds: None,
+        }) => last_reset_timestamp_seconds);
+
+        // Resetting the timers cannot be done sooner than `RESET_TIMERS_COOL_DOWN_INTERVAL` after
+        // the canister is initialized.
+        state_machine.advance_time(Duration::from_secs(1000));
+        state_machine.tick();
+
+        let last_spawned_timestamp_seconds = assert_matches!(get_timers(), Some(governance_pb::Timers {
+            last_reset_timestamp_seconds: Some(last_reset_timestamp_seconds_1),
+            last_spawned_timestamp_seconds: Some(last_spawned_timestamp_seconds),
+        }) => {
+            assert_eq!(last_reset_timestamp_seconds_1, last_reset_timestamp_seconds);
+            last_spawned_timestamp_seconds
+        });
+
+        assert_eq!(
+            last_spawned_timestamp_seconds,
+            last_reset_timestamp_seconds + 1000
+        );
+        last_spawned_timestamp_seconds
+    };
+
+    // Reset the timers.
+    {
+        let payload = Encode!(&swap_pb::ResetTimersRequest {}).unwrap();
```

Comment:
I might extract "reset_timers" into a functionm as well as "get_timers",

Reply:
Will do this in a follow-up to avoid invalidating the CI pipeline.

Reply:
Done in https://github.com/dfinity/ic/pull/2214